### ISSUE TITLE
nw does not support by default ES6 features that iojs does support by default

### ISSUE THREAD

**Issue by vsemozhetbyt** [nwjs/nw.js#3045]
According to https://iojs.org/es6.html I can use in iojs, for example, template strings without any flag. However nw 0.12.0-alpha2 does not support them by default (Uncaught SyntaxError: Unexpected token ILLEGAL). Is this behaviour consistent?

**Comment by vsemozhetbyt** [comment]
Well, "js-flags": "--harmony" in the Manifest changes nothing. What should I do for template strings support in the nw?

**Comment by ITBeka** [comment]
It seems that nw is based on node.js > 0.12.0 not on io.js

**Comment by vsemozhetbyt** [comment]
Well, 0.12.0-alpha2 gives "node": "1.0.0" in the process.versions. Does not this mean iojs?

**Comment by ITBeka** [comment]
So your issue shows that your nw uses node. while here https://groups.google.com/forum/#!topic/nwjs-general/V1FhvfaFIzQ Roger Wang says that it is io.js in alpha1 release.
So I am a bit confused now.

**Comment by vsemozhetbyt** [comment]
I think that "node" is just a temporal alias for iojs here. The version is more significant sign.

**Comment by ITBeka** [comment]
I think you are right. I have no possibility to make my hands dirty on this to figure all out.
Anyway всеможетбыть))

**Comment by ITBeka** [comment]
Fortunately I found nw in one of my projects. and typing in toolbar nw:version shows
nw.js v0.12.0-alpha2
io.js v1.0.0
Chromium 41.0.2236.2
commit hash: e0d5ce6-d017875-34492e5-2e978ac-1116f2c-fd87c8d

So maybe you should check whether you are using right version of nw.
Maybe update will fix it.

**Comment by vsemozhetbyt** [comment]
I have completely the same info except "node".

**Comment by mrfabbri** [comment]
Confirmed on nwjs-v0.12.0-alpha2 (Mac OS X 10.10 Yosemite 64 bit). Running:

````javascript
var ANSWER = 42;
console.log(`The answer to life the universe and everything is ${ANSWER} `);
````

would produce a `SyntaxError: Unexepected token ILLEGAL`

@vsemozhetbyt I think it's an issue on the V8 side (string templates where added recently, v8 on nwjs is v3.31.31 while in iojs is v3.31.74.1). The other ES6 features (e.g. Promises, Map, WeakMap… and even generators) work fine. 

/cc @rogerwang

**Comment by vsemozhetbyt** [comment]
process.versions in iojs-v1.0.4 gives: "node": "1.0.4", "v8": "4.1.0.12".

**Comment by mrfabbri** [comment]
yes, that's io.js v1.0.4, I was comparing nwjs v0.12.0-alpha2 with io.js v1.0.0 (the one currently shipped with nwjs v0.12.0-alpha2)

**Comment by rogerwang** [comment]
The the git tip has been updated to io.js v1.0.4. It will be released with in the next version (alpha3).
